src/GridList/GridList.js

```diff
@@ -0,0 +1,119 @@
+// @flow weak
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { createStyleSheet } from 'jss-theme-reactor';
+import customPropTypes from '../utils/customPropTypes';
+
+export const styleSheet = createStyleSheet('MuiGridList', () => {
+  return {
+    root: {
+      display: 'flex',
+      flexWrap: 'wrap',
+      overflowY: 'auto',
+    },
+    item: {
+      boxSizing: 'border-box',
+    },
+  };
+});
+
+/**
+ * ```jsx
+ * <GridList>
+ *   <GridTile>
+ *     <img src="image.jpg" />
+ *     <GridTileTitlebar title="GridTile" />
+ *   </GridTile>
+ * </GridList>
+ * ```
+ */
+export default function GridList(props, context) {
+  const {
+    cols,
+    padding,
+    cellHeight,
+    children,
+    className: classNameProp,
+    component: ComponentProp,
+    ...other
+  } = props;
+
+  const classes = context.styleManager.render(styleSheet);
+
+  const wrappedChildren = React.Children.map(children, (currentChild) => {
+    if (React.isValidElement(currentChild) && currentChild.type.muiName === 'ListSubheader') {
+      return currentChild;
+    }
+    const childCols = currentChild.props.cols || 1;
+    const childRows = currentChild.props.rows || 1;
+    const itemStyle = {
+      width: `${((100 / cols) * childCols)}%`,
+      height: cellHeight === 'auto' ? 'auto' : (cellHeight * childRows) + padding,
+      padding: padding / 2,
+    };
+
+    return <div className={classes.item} style={itemStyle}>{currentChild}</div>;
+  });
+
+  return (
+    <ComponentProp
+      className={classNames(classes.root, classNameProp)}
+      style={{ margin: -padding / 2 }}
```

Comment:
Same point than last time, we don't take into account the style provided by users.